new readme and documentation (#106)
kioku-project/kioku: README.md

![kioku](https://github.com/kioku-project/kioku/assets/60541979/1f827df3-5882-4285-913f-47f04b26196b)

![GitHub contributors](https://img.shields.io/github/contributors/kioku-project/kioku)
[![Backend Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=kioku-project_kioku_services&metric=alert_status)](https://sonarcloud.io/summary/new_code?id=kioku-project_kioku_services)
[![Frontend Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=kioku-project_kioku_frontend&metric=alert_status)](https://sonarcloud.io/summary/new_code?id=kioku-project_kioku_frontend)
[![Frontend Vulnerabilities](https://sonarcloud.io/api/project_badges/measure?project=kioku-project_kioku_frontend&metric=vulnerabilities)](https://sonarcloud.io/summary/new_code?id=kioku-project_kioku_frontend)
[![Storybook link](https://github.com/storybookjs/brand/blob/master/badge/badge-storybook.svg)](https://main--63e354941aa15501d3467f88.chromatic.com)

# Welcome to Kioku!

> The cloud native flashcard application that focusses on collaborative content creation.

Reviewer: ```suggestion
The cloud native flashcard application that focuses on collaborative content creation.
```

Author: implemented in ee4f7a25

  <img alt="Available icon" src="">
</picture>
  <b>Available</b>

Access your flashcards everywhere and at any time. Switch seamlessly between multiple platforms and never miss a learning opportunity again!
</td>
<td>
    <picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github.com/kioku-project/kioku/assets/60541979/76142857-6749-419a-becc-86b234f16d42">
  <source media="(prefers-color-scheme: light)" srcset="https://github.com/kioku-project/kioku/assets/60541979/87bfe7ed-e292-4cd1-a6a3-ac26c0ea27ca">
  <img alt="Entertaining icon" src="">
</picture>
  <b vertical-align="center">Entertaining</b>

Achievements and leaderboards make learning more engaging and motivating. Kioku helps you to achieve better results and stay on track with your personal learning goals!
</td>
</tr>
</tbody>
</table>

# Index
1. [Getting started](#getting-started)
2. [Frontend development](./doc/frontend_development.md)
3. [Backend development](./doc/backend_development.md)
4. [Deployment](./doc/deployment.md)

# Getting started
In order to run Kioku locally, first clone the repository
```
git clone https://github.com/kioku-project/kioku
```

> move into the `backend` folder and configure the .env file

Reviewer 2: ```suggestion
move into the `backend` and `frontend` folder and configure both .env files
```

Reviewer 2: Okay, then I would suggest to add information about the `.env` file in the frontend_development file of the documentation.

Author: implemented in ee4f7a25

> move into the `backend` folder and configure the .env file

Reviewer: and configure the `.env` file in the `backend` folder.

Author: implemented in 0dafe9e4

```
cp .env.example .env
```

> [!WARNING]
> The example environment file is populated with default values, be sure to change all values before using the application in production!

> finally, start frontend and backend using docker compose

Reviewer: ```suggestion
Finally, start frontend and backend using docker compose
```

Author: implemented in ee4f7a25

[![Backend Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=kioku-project_kioku_services&metric=alert_status)](https://sonarcloud.io/summary/new_code?id=kioku-project_kioku_services)
[![Frontend Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=kioku-project_kioku_frontend&metric=alert_status)](https://sonarcloud.io/summary/new_code?id=kioku-project_kioku_frontend)
[![Backend Vulnerabilities](https://sonarcloud.io/api/project_badges/measure?project=kioku-project_kioku_services&metric=vulnerabilities)](https://sonarcloud.io/summary/new_code?id=kioku-project_kioku_services)
[![Frontend Vulnerabilities](https://sonarcloud.io/api/project_badges/measure?project=kioku-project_kioku_frontend&metric=vulnerabilities)](https://sonarcloud.io/summary/new_code?id=kioku-project_kioku_frontend)
[![Storybook link](https://github.com/storybookjs/brand/blob/master/badge/badge-storybook.svg)](https://main--63e354941aa15501d3467f88.chromatic.com)

# Welcome to Kioku!

The cloud native flashcard application that focuses on collaborative content creation.

## Features

<table>
<tbody>
<tr>
<td><picture>

>   <source media="(prefers-color-scheme: dark)" srcset="https://github.com/kioku-project/kioku/assets/60541979/0cc0f108-f0e7-49c5-bafc-02123afdf514">

Reviewer: ```suggestion
  <source media="(prefers-color-scheme: dark)" srcset="https://github.com/kioku-project/kioku/assets/71837281/62942574-34ef-4e75-9b17-9721444f9d74">
```

  <img alt="Collaborative icon" src="">
</picture>
  <b>Collaborative</b>

Collaborate with your friends and fellow students in groups and work on shared decks. Learn together and motivate each other!

</td>
<td><picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github.com/kioku-project/kioku/assets/60541979/a3dbfa9e-0b38-477d-b1eb-705f30b45eda">
  <source media="(prefers-color-scheme: light)" srcset="https://github.com/kioku-project/kioku/assets/60541979/91a42bb7-8985-435f-8c61-c8a1f455dd7c">
  <img alt="Individual icon" src="">
</picture>
  <b>Individual</b>

Create and customize your own flashcards tailored to your needs and preferences. Set your own pace with our spaced repetition system to maximize your potential!

</td>
</tr>
<tr>
<td><picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github.com/kioku-project/kioku/assets/60541979/d2753ef0-6e62-48f5-9b1c-fc4aea450a1b">
  <source media="(prefers-color-scheme: light)" srcset="https://github.com/kioku-project/kioku/assets/60541979/f18f855c-6f26-49a7-84eb-18ef94838c69">
  <img alt="Compatible icon" src="">
</picture>
  <b>Compatible</b>

> Kioku is compatible with Anki, allowing you to import and export your existing decks into our application while taking advantage of Kioku's collaborative features!

Reviewer: ```suggestion
Kioku is compatible with [Anki](https://apps.ankiweb.net/index.html), allowing you to import and export your existing decks into our application while taking advantage of Kioku's collaborative features!
```

</picture>
  <b>Available</b>

Access your flashcards everywhere and at any time. Switch seamlessly between multiple platforms and never miss a learning opportunity again!

</td>
<td>
    <picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github.com/kioku-project/kioku/assets/60541979/76142857-6749-419a-becc-86b234f16d42">
  <source media="(prefers-color-scheme: light)" srcset="https://github.com/kioku-project/kioku/assets/60541979/87bfe7ed-e292-4cd1-a6a3-ac26c0ea27ca">
  <img alt="Entertaining icon" src="">
</picture>
  <b vertical-align="center">Entertaining</b>

Achievements and leaderboards make learning more engaging and motivating. Kioku helps you to achieve better results and stay on track with your personal learning goals!

</td>
</tr>
</tbody>
</table>

# Index

1. [Getting started](#getting-started)
2. [Frontend development](./doc/frontend_development.md)
3. [Backend development](./doc/backend_development.md)
4. [Deployment](./doc/deployment.md)

# Getting started

In order to run Kioku locally, first clone the repository

```
git clone https://github.com/kioku-project/kioku
```

and configure the `.env` file in the `backend` folder

```

> cp .env.example .env

Reviewer: ```suggestion
cp .env{.example,}
```

```

> [!WARNING]
> The example environment file is populated with default values, be sure to change all values before using the application in production!

> Finally, start frontend and backend using docker compose

Reviewer: ```suggestion
Finally, start the frontend and backend using [Docker Compose](https://docs.docker.com/compose/).
```

```
docker compose up
```

> after that, the frontend is reachable at `http://localhost:3000`  
> to reach pgAdmin, open `http://localhost:3002`  
> the micro dashboard is reachable at `http://localhost:3001`

Reviewer: ```suggestion
|Service|URL|
|---|---|
| **Frontend** | [`http://localhost:3000`](http://localhost:3000) |
| **pgAdmin** | [`http://localhost:3002`](http://localhost:3002) |
| **micro dashboard** | [`http://localhost:3001`](http://localhost:3001) |
```

Access your flashcards everywhere and at any time. Switch seamlessly between multiple platforms and never miss a learning opportunity again!

</td>
<td>
    <picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github.com/kioku-project/kioku/assets/60541979/76142857-6749-419a-becc-86b234f16d42">
  <source media="(prefers-color-scheme: light)" srcset="https://github.com/kioku-project/kioku/assets/60541979/87bfe7ed-e292-4cd1-a6a3-ac26c0ea27ca">
  <img alt="Entertaining icon" src="">
</picture>
  <b vertical-align="center">Entertaining</b>

Achievements and leaderboards make learning more engaging and motivating. Kioku helps you to achieve better results and stay on track with your personal learning goals!

</td>
</tr>
</tbody>
</table>

# Index

1. [Getting started](#getting-started)
2. [Frontend development](./doc/frontend_development.md)
3. [Backend development](./doc/backend_development.md)
4. [Deployment](./doc/deployment.md)

# Getting started

In order to run Kioku locally, first clone the repository

```
git clone https://github.com/kioku-project/kioku
```

and configure the `.env` file in the `backend` folder

```
cp .env.example .env
```

> You can generate the JWT_PRIVATE_KEY with the following command

Reviewer: ```suggestion
You can generate the `JWT_PRIVATE_KEY` with the following command
```
